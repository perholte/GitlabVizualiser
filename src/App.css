#graph-container > svg {
	width: fit-content;
	height: fit-content;
}

#graph-container {
	padding-top: 40px;
	max-width: 80%;	
	max-height: 600px;
	border-color: black;
}
