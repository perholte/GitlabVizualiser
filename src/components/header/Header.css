/* .header-btn {
    margin: auto 2em;
    width: auto;
    flex-direction: column;
    width: auto;
}

.header-btn>button {
    width: 100%;
} */
