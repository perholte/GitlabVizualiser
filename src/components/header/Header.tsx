import { Flex, Button } from '@chakra-ui/react';
import { Link } from 'react-router-dom';

const Header = () => {
	return (
		<>
			<Flex
				as='nav'
				align='center'
				justify='space-evenly'
				w='100%'
				mb={0}
				p={8}
				bg={'seagreen'}
				color={'MenuText'}
				minH='15vh'
				maxH='15vh'
			>
				<Link to='/'>
					<Button marginX='1rem' h='3rem' w='full'>
						Home
					</Button>
				</Link>
				<Link to='/branches'>
					<Button marginX='1rem' h='3rem' w='full'>
						Branches
					</Button>
				</Link>
				<Link to='/issues'>
					<Button marginX='1rem' h='3rem' w='full'>
						Issues
					</Button>
				</Link>
				<Link to='/contributors'>
					<Button marginX='1rem' h='3rem' w='full'>
						Contributors
					</Button>
				</Link>
				<Link to='/messages'>
					<Button marginX='1rem' h='3rem' w='full'>
						Commits
					</Button>
				</Link>
			</Flex>
		</>
	);
};

export default Header;
