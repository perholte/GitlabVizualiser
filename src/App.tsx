import React, { useEffect } from 'react';
import { useState } from 'react';
import { Route, Switch } from 'react-router-dom';
import './App.css';
import BranchTree from './components/branchTree/BranchTree';
import CommitMessages from './components/commitMessages/CommitMessages';
import Contributions from './components/contributions/Contributions';
import IssueList from './components/issues/IssueList';
import Header from './components/header/Header';
import Frontpage from './components/frontpage/Frontpage';

export const ThemeContext = React.createContext({ darkmode: false });

function App() {
	const [darkmode, setDarkmode] = useState<boolean>(() =>
		JSON.parse(localStorage.getItem('darkmode') ?? 'false')
	);

	useEffect(() => {
		localStorage.setItem('darkmode', JSON.stringify(darkmode));
	}, [darkmode]);

	return (
		<ThemeContext.Provider value={{ darkmode: darkmode }}>
			<Header handleToggleTheme={() => setDarkmode(!darkmode)} />
			<Switch>
				<Route path='/branches'>
					<BranchTree />
				</Route>
				<Route path='/issues'>
					<IssueList />
				</Route>
				<Route path='/Contributors'>
					<Contributions />
				</Route>
				<Route path='/messages'>
					<CommitMessages />
				</Route>
				<Route path='/'>
					<Frontpage />
				</Route>
			</Switch>
		</ThemeContext.Provider>
	);
}

export default App;
