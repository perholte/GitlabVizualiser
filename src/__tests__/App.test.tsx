import { cleanup, render, screen } from '@testing-library/react';
import App from '../App';
import { BrowserRouter } from 'react-router-dom';
import { act } from 'react-dom/test-utils';
import { QueryClient, QueryClientProvider } from 'react-query';
import Header from '../components/header/Header';
import userEvent from '@testing-library/user-event';

// Prosjektet skal vise oppsett av og eksempel på testing med Jest - minimum er å ha en snapshottest og noen enkle tester på komponentenes oppførsel.

beforeEach(() => {
	return render(
		<QueryClientProvider client={new QueryClient()}>
			<BrowserRouter>
				<App />
			</BrowserRouter>
		</QueryClientProvider>
	);
});
afterEach(cleanup);

it('renders the App correctly', () => {
	expect(<App></App>).toMatchSnapshot();
});

it('Renders the header correctly', () => {
	expect(
		<Header
			handleToggleTheme={function () {
				throw new Error('Function not implemented.');
			}}
		></Header>
	).toMatchSnapshot();
});

describe('testing implementation and representation of sessionstorage', () => {
	it('Writes a string in the input field and submits it', () => {
		window.location.pathname = '/';
		const inputLabel = screen.getByRole('textbox');
		const button = screen.getByText('Submit name');
		userEvent.type(inputLabel, 'test');
		button.click();
		expect(sessionStorage.getItem('name')).toEqual('test');
	});
});

describe('Checking the routing of the header', () => {
	it('Should be on the home page', () => {
		const homeLink = screen.getByText('Home');
		homeLink.click();
		expect(window.location.pathname).toEqual('/');

		const input = screen.getByRole('checkbox')
		expect(input).toBeInTheDocument()
	});

	it('Re-routes to branches when branches-link is clicked', async () => {
		act(() => {
			const branchLink = screen.getByText('Branches');
			branchLink.click();
		});
		expect(window.location.pathname).toEqual('/branches');

		const svg = document.getElementById('graph-container');
		expect(svg).toBeInTheDocument();
	});

	it('Re-routes to commit-messages when commit-link is clicked', async () => {
		act(() => {
			const commitLink = screen.getByText('Commits');
			commitLink.click();
		});
		expect(window.location.pathname).toEqual('/messages');
	});
	//Check that the correct component is showed.

	it('Re-routes to contributors when contributor-link is clicked', async () => {
		act(() => {
			const contributorLink = screen.getByText('Contributors');
			contributorLink.click();
		});
		expect(window.location.pathname).toEqual('/contributors');
	});
	//Check that the correct component is showed.
});
