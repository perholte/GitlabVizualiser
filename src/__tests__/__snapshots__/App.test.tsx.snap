// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Renders the header correctly 1`] = `
<Header
  handleToggleTheme={[Function]}
/>
`;

exports[`renders the App correctly 1`] = `<App />`;
